.github/api_best_practices.md

```diff
@@ -0,0 +1,198 @@
+# ACS API Guidelines
+An overview of the set of standardized guidelines and practices to ensure that the APIs across different 
+projects and teams adhere to a common structure and design. This consistency simplifies the process of understanding 
+and maintaining APIs. The guidelines helps identify potential pitfalls and design choices that could lead to issues to 
+help developers avoid common mistakes and build more robust APIs. When in doubt, It is encouraged to seek resolution
+in slack channel [#forum-acs-api-design](https://redhat-internal.slack.com/archives/C05MMG2PP8A).
+
+## Table of Contents
+
+- [Naming Guidelines](#naming-guidelines)
+  - [Service Name](#service-name)
+  - [Method Name](#method-name)
+  - [Message Name](#message-name)
+
+## Naming Guidelines
+
+### Service Name
+Service name **must** be unique and should use a noun that generally refers to a resource or product component 
+e.g. `DeploymentService`, `ReportService`, `ComplianceService`. Intuitive and well-known short forms or 
+abbreviations may be used in some cases (and could even be preferable) for succinctness 
+e.g. `ReportConfigService`, `RbacService`. 
+All RPC methods grouped into a single service must generally pertain to the primary resource of the service.
+
+All service defined in versioned package `/proto/api/v2` **must not** import from 
+`/proto/storage` and `/proto/internalapi` packages.
+All new RPC methods defined in versioned package `/proto/api/v1` **must not** import from
+`/proto/storage` and `/proto/internalapi` packages.
+
+### Method Name
+Methods (and URLs) should be named such that they provide insights into the functionality. Typically, the method 
```

Comment:
This is the only mention of URLs in this doc.

I would like us to specify that URLs should use hyphens to separate words in the URL.

It makes it easier to parse the URL quickly, and also reduces the chance of typos for users who are using the API at the command line.

You can see this in the GitHub API here, with the `dependency-graph` part of the URL,
https://docs.github.com/en/rest/dependency-graph/dependency-review?apiVersion=2022-11-28#get-a-diff-of-the-dependencies-between-commits

It's also part of Google's URL style guide for human-readable URLs,
https://developers.google.com/search/docs/crawling-indexing/url-structure

Reply:
@vjwilson good catch. The section for URL naming will be separate. I missed removing this one reference to URLs.